colin-api/src/colin_api/resources/reset.py

```diff
@@ -55,3 +56,41 @@ def post():
             # general catch-all exception
             current_app.logger.error(err.with_traceback(None))
             return jsonify({'message': 'Error when trying to reset COLIN'}), 500
+
+
+@cors_preflight('POST')
+@API.route('/by_event_id')
+class ResetByEventId(Resource):
+    """Reset filing(s) based on the provided event_id, or array of event_ids.
+      This is only tested to work on Annual Reports, ymmv"""
+
+    eventResetParser = API.parser()
+    eventResetParser.add_argument(
+        'event_ids',
+        type=list,
+        help='The list of event ids to reset. Can be one id',
+        location='json',
+        required=True)
+
+    @staticmethod
+    @cors.crossdomain(origin='*')
+    @jwt.requires_roles([COLIN_SVC_ROLE])
+    @API.expect(eventResetParser)
+    def post():
+        """Reset filing(s) based on the provided event_id, or array of event_ids.
+        This is only tested to work on Annual Reports, ymmv"""
+        try:
+            
+            event_ids = API.payload.get('event_ids', None)
+
+            for event in event_ids:
+                Reset.reset_filings_by_event(
+                  event_ids=event_ids
+                )
```

Comment:
is this for loop needed?  

It looks like ` Reset.reset_filings_by_event(event_ids=event_ids)` can handle `event_ids` as a list already?

Reply:
Good catch